### PR TITLE
Hide stuff requiring licensees you don't have in the map sales panels

### PR DESCRIPTION
**Pull request by warp-core**
**Feature:** This PR implements the feature request from @Zitchas on Discord.

## Feature Details
Adds the option to toggle not showing outfits and ships you do not have the licenses for with the 'l' key in the map shipyard or outfitter panels.

Setting does not currently persist across switching between different map views or closing and reopening the map altogether.
- Should that state be pilot specific or global (stored in preferences)?

## UI Screenshots
all | only what I have licenses for
-- | --
image | image

## Usage Examples
Press the 'l' ('L') key while in one of the map sales panels.

## Testing Done
Open the map sales panels, press the 'l' key, see that the items I don't have licenses for are removed from the list, then re-added when I press the key again.

### Automated Tests Added
N/A

## Performance Impact
N/A

### CODE REVIEW

File: source/MapOutfitterPanel.h [hunk 1]
```diff
@@ -51,9 +51,7 @@ class MapOutfitterPanel : public MapSalesPanel {
 
 	virtual void DrawItems() override;
 
-
-private:
-	void Init();
+	virtual void Init() override;
```
Comment (Hurleveur) [inline source/MapOutfitterPanel.h:54]: Why did you change stuff with the init methods? Seems unrelated to me?

**Review by Hurleveur — COMMENTED** [review]
Yeees, moar hidden features
Mb we should add a small license somewhere that you can click on to trigger this as well and is a visual indicator? Could always be done later on ig

**Comment by warp-core** [comment]
> Yeees, moar hidden features
> Mb we should add a small license somewhere that you can click on to trigger this as well and is a visual indicator? Could always be done later on ig

I'll add something about it to a help message.

File: source/MapOutfitterPanel.h (hunk 1 above)
Comment (warp-core) [reply source/MapOutfitterPanel.h:54]: It's so I can call `Init()` from `MapSalesPanel::KeyDown()` to recompile the catalog when the 'l' key is pressed.

Comment (Hurleveur) [reply source/MapOutfitterPanel.h:54]: oh right, that's ok I think since it doesnt happen that often and we'd do the same thing anyway if we did a separate method

**Review by Hurleveur — APPROVED** [review]
looks alr, would still be nice to get it to have a visual element tho, eventually

**Comment by quyykk** [comment]
IMO @Saugia should make it so that the Gegno ships don't appear (or at least, not all of them) in the list. Mainly because the UX here is not great

**Comment by Saugia** [comment]
Yeah, I do that in the Geghancement PR.
